homework/index.js

```diff
@@ -29,23 +31,160 @@
+      text: repo.forks,
+      class: 'texts',
+    });
+
+    createAndAppend('span', updatedTime, {
+      text: convertDateTime(repo.updated_at),
+      class: 'texts',
+    });
+  }
+
+  function getContributorsInfo(repositories, ul) {
+    ul.textContent = ' ';
+    const url = repositories.contributors_url;
+    fetchJSON(url)
+      .then(contributions => {
+        contributions.forEach(contributor => {
+          const li = createAndAppend('li', ul, { class: 'contributors-li' });
+          const contributorImage = createAndAppend('div', li, {
+            class: 'contributor-image',
+          });
+          createAndAppend('img', contributorImage, {
+            src: contributor.avatar_url,
+            alt: contributor.login,
+          });
+
+          const contributorName = createAndAppend('div', li, {
+            class: 'contributor-name',
+          });
+          createAndAppend('a', contributorName, {
+            text: contributor.login,
+            href: contributor.html_url,
+            target: '_blank',
+          });
+
+          createAndAppend('span', contributorName, {
+            text: contributor.contributions,
+            class: 'added-contributions',
+          });
+        });
+      })
+      .catch(err => {
+        createAndAppend('div', document.getElementById('root'), {
+          text: err.message,
+          class: 'alert-error',
+        });
+      });
   }
 
   function main(url) {
-    fetchJSON(url, (err, repos) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, {
+    fetchJSON(url)
+      .then(repos => {
+        const root = document.getElementById('root');
+        const headerDiv = createAndAppend('header', root, {
+          text: 'HYF Repositories',
+          class: 'repo-header',
+        });
+
+        const select = createAndAppend('select', headerDiv, {
+          class: 'select-repository',
+        });
+        repos.sort(sortRepositoriesByName).forEach((repo, index) =>
+          createAndAppend('option', select, {
+            text: repo.name,
+            value: index,
+          }),
+        );
+
+        const mainContainer = createAndAppend('main', root, {
+          class: 'main-container',
+        });
+
+        const repositoryContainer = createAndAppend('section', mainContainer, {
+          class: 'repository-section',
+        });
+
+        const ul = createAndAppend('ul', repositoryContainer, {
+          class: 'ul',
+        });
+
+        renderRepoDetails(repos[0], ul);
+        const contributorContainer = createAndAppend('section', mainContainer, {
+          class: 'contributors-section',
+        });
+
+        createAndAppend('h2', contributorContainer, {
+          text: 'Contributors',
+        });
+
+        const ulContributors = createAndAppend('ul', contributorContainer);
+        getContributorsInfo(repos[0], ulContributors);
+
+        select.addEventListener('click', () => {
```

Comment:
You don't want this event handler to fire whenever this user clicks on the select box. It should fire only when the user changes the selection. The event to list for is `change`.